apps/heft/src/cli/actions/BuildAction.ts

```diff
@@ -93,7 +178,54 @@ export class BuildAction extends HeftActionBase<IBuildActionData, BuildHooks> {
   }
 
   protected async actionExecute(actionData: IBuildActionData): Promise<void> {
-    throw new Error('Not implemented yet...');
+    if (this._cleanFlag.value) {
+      await this._runWithLogging('Clean', async () => await this._cleanAction.executeInner());
+    }
+
+    const preCompilePhase: IBuildPhase = { hooks: new BuildPhaseHooksBase() };
+    actionData.hooks.preCompile.call(preCompilePhase);
+
+    const compilePhase: ICompilePhase = {
+      hooks: new CompilePhaseHooks(),
+      copyStaticAssetsConfiguration: {
+        fileExtensions: [],
+        exclude: [],
+        include: [],
+        sourceFolderName: undefined,
+        destinationFolders: []
+      }
+    };
+    actionData.hooks.compile.call(compilePhase);
+
+    const bundlePhase: IBundlePhase = { hooks: new BuildPhaseHooksBase() };
+    actionData.hooks.bundle.call(bundlePhase);
+
+    const postBuildPhase: IBuildPhase = { hooks: new BuildPhaseHooksBase() };
+    actionData.hooks.postBuild.call(postBuildPhase);
+
+    if (actionData.watchMode) {
```

Comment:
This is fine for now. 

From the design discussions about watch, the consensus was that ideally Rush should centrally watch the entire monorepo, rather than each Heft task watching its own files.  This would reduce the amount of watcher overhead.  And also, when a project depends on inputs from an upstream project, Rush will be notified when the upstream project has signaled completion.  Whereas Heft can only see that its input file has been touched -- which might be a false alarm due to half-written output from an upstream process.

Reply:
Yeah that's a model that we should eventually move to, but that will involve a lot of work on the Rush side before we can start using it here.